src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php

```diff
@@ -61,4 +75,68 @@ public function configureOptions(OptionsResolver $resolver): void
             'data_class' => HeidelpayPaymentTransfer::class,
         ])->setRequired(static::OPTIONS_FIELD_NAME);
     }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $this->addDateOfBirth($builder);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addDateOfBirth(FormBuilderInterface $builder)
+    {
+        $builder->add(
+            static::FORM_FIELD_DATE_OF_BIRTH,
+            BirthdayType::class,
+            [
+                'label' => static::LABEL_DATE_OF_BIRTH,
+                'required' => true,
+                'widget' => static::WIDGET_TYPE,
+                'format' => static::FORMAT_DATE_OF_BIRTH,
+                'input' => static::INPUT_TYPE,
+                'attr' => [
+                    'placeholder' => static::PLACEHOLDER_DATE_OF_BIRTH,
+                ],
+
+                'constraints' => [
+                    $this->createNotBlankConstraint(),
+                    $this->createBirthdayConstraint(),
+                ],
+            ]
+        );
+
+        return $this;
+    }
+
+    /**
+     * @return \Symfony\Component\Validator\Constraint
+     */
+    protected function createNotBlankConstraint(): Constraint
+    {
+        return new NotBlank(['groups' => $this->getPropertyPath()]);
+    }
+
+    /**
+     * @return \Symfony\Component\Validator\Constraint
+     */
+    protected function createBirthdayConstraint(): Constraint
+    {
+        return new Callback([
+            'callback' => function ($date, ExecutionContextInterface $context) {
+                if (strtotime($date) > strtotime(self::MIN_BIRTHDAY_DATE_STRING)) {
```

Comment:
Why not `>=`?
Also I would prefer to work with dates only via `DateTime` classes

Reply:
Refactored to use `DateTime` object.
`>=` will affect customer that has turned 18 today.